### PR TITLE
Prevent NumGridPointsAndGridSpacing element dist when using GTS

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Also prevent using NumGridPointsAndGridSpacing in the elliptic code as well.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/ElementDistribution.hpp [hunk 1]
```diff
@@ -173,15 +176,51 @@ struct BlockZCurveProcDistribution {
...
+};
+
+template <typename Metavariables>
+struct uses_lts<Metavariables, false> {
+  static constexpr bool value = false;
+};
+
+template <typename Metavariables>
+static constexpr bool uses_lts_v =
+    uses_lts<Metavariables, has_local_time_stepping_v<Metavariables>>::value;
```
Comment (wthrowe) [inline src/Domain/ElementDistribution.hpp]: This is `CREATE_GET_STATIC_MEMBER_VARIABLE_OR_DEFAULT`.

File: src/Domain/ElementDistribution.hpp [hunk 2]
```diff
@@ -173,15 +176,51 @@ struct BlockZCurveProcDistribution {
...
+    const auto ordering = options.parse_as<std::string>();
+    if (ordering == "Uniform") {
+      return domain::ElementWeight::Uniform;
+    } else if (ordering == "NumGridPoints") {
+      return domain::ElementWeight::NumGridPoints;
+    } else if (ordering == "NumGridPointsAndGridSpacing") {
+      if constexpr (not element_weight_detail::uses_lts_v<Metavariables>) {
+        PARSE_ERROR(
+            options.context(),
+            "When using global time stepping, you cannot use "
```
Comment (wthrowe) [inline src/Domain/ElementDistribution.hpp]: "using GTS" -> "not using LTS" to include elliptic.

File: src/Domain/Tags/ElementDistribution.hpp [hunk 3]
```diff
@@ -35,6 +35,15 @@ struct ElementDistribution {
...
 namespace Tags {
+/// \ingroup DataBoxTagsGroup
+/// \ingroup ComputationalDomainGroup
+/// Tag that holds method for how to distribute the elements on the given
+/// resources.
+///
+/// \note When using global time stepping (GTS), a user cannot choose the
+/// NumGridPointsAndGridSpacing element distribution because grid spacing does
+/// not affect the computational cost at all. Therefore, if a user does choose
+/// NumGridPointsAndGridSpacing when using GTS, an error will occur.
```
Comment (wthrowe) [inline src/Domain/Tags/ElementDistribution.hpp]: Again, GTS -> not LTS

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by knelli2** [comment]
I squashed directly

**Review by wthrowe — APPROVED** [review]

**Pull request merged**